src/main/groovy/org/asciidoctor/groovydsl/AsciidoctorExtensions.groovy

```diff
@@ -46,26 +46,39 @@ class AsciidoctorExtensions {
 
     static void registerTo(Asciidoctor asciidoctor) {
         AsciidoctorExtensionHandler extensionHandler = new AsciidoctorExtensionHandler(asciidoctor)
-        REGISTERED_EXTENSIONS.each {
+        for (def it : REGISTERED_EXTENSIONS) {
             DelegatingScript script
             switch (it) {
                 case Closure:
-                    it.delegate = extensionHandler
-                    it.call()
+                    try {
+                        it.delegate = extensionHandler
+                        it.call()
+                    } catch (e) {
+                        REGISTERED_EXTENSIONS.clear()
+                        throw new AsciidoctorExtentionException("Error registering extension from class in ${it.class.name}", e)
```

Comment:
Codenarc doesn't like the GString here.
Could you change the `"Error..."` to `'Error...'`?